src/Raven.Server/Rachis/Leader.cs

```diff
@@ -682,10 +699,17 @@ private void EmptyQueue()
                 {
                     while (_commandsQueue.TryDequeue(out var cmd))
                     {
```

Comment:
This will run forever if we a lot of incoming commands (cluster transactions for example).
It should be limited by the number of transactions or the memory size.